library/common/extensions/filters/http/platform_extension/bridging_filter.cc

```diff
@@ -0,0 +1,115 @@
+#include "library/common/extensions/filters/http/platform_extension/bridging_filter.h"
+
+//#include <iostream>
+
+#include "envoy/server/filter_config.h"
+
+#include "common/common/assert.h"
+#include "common/common/utility.h"
+
+#include "library/common/api/external.h"
+#include "library/common/buffer/utility.h"
+#include "library/common/http/header_utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace PlatformExtension {
+
+Http::FilterHeadersStatus mapStatus(envoy_filter_headers_status_t status) {
+  switch (status) {
+  case ENVOY_FILTER_HEADERS_STATUS_CONTINUE:
+    return Http::FilterHeadersStatus::Continue;
+  case ENVOY_FILTER_HEADERS_STATUS_STOP_ITERATION:
+    return Http::FilterHeadersStatus::StopIteration;
+  case ENVOY_FILTER_HEADERS_STATUS_CONTINUE_AND_END_STREAM:
+    return Http::FilterHeadersStatus::ContinueAndEndStream;
+  case ENVOY_FILTER_HEADERS_STATUS_STOP_ALL_ITERATION_AND_BUFFER:
+    return Http::FilterHeadersStatus::StopAllIterationAndBuffer;
+  default:
+    ASSERT(false, "unrecognized filter status from platform: {}");
+    return Http::FilterHeadersStatus::Continue;
+  }
+}
+
+BridgingFilterConfig::BridgingFilterConfig(
+    const envoymobile::extensions::filters::http::platform_extension::Bridging& proto_config)
+    : name_(proto_config.name()) {}
+
+BridgingFilter::BridgingFilter(BridgingFilterConfigSharedPtr config)
+    : platform_filter_(
```

Comment:
I wonder if this should live in the BridgingFilterConfig class. That way if we crash/throw an exception we can do so at config time and not when the first stream creates the filter.

Reply:
Remember that for the current use case, with dynamic filter configuration, this will be handled correctly behind the scenes, at config time.

For the general case I don't really know how to make that happen without significant new plumbing, because the platform API is a runtime-defined memory location, that must be passed by the platform down into Envoy. This would require threading that into Envoy's startup sequencing. I'd suggest it could maybe be an item for future work, but a crash an API lookup seemed good enough for now (again, especially since dynamic configuration will handle this correctly without user intervention).

Reply:
Ok, my thinking was that a user would register platform filters after engine instantiation but before initialization. So then it would make sense to have the lookup in the config. But if the thinking is that a user might register filters after engine initialization this makes sense. 

So then, I guess the larger question is how long do we want to support filter registration.

Reply:
I think we're misunderstanding each other. I am indeed planning to register filters prior to Engine initialization.

But I think I understand you now. You're suggesting the config could perform the API lookup during initialization. Sure - in fact, I agree, I like that better. (Is that what you mean?)